### PR TITLE
Natalieparry/fixdllbuild

### PR DESCRIPTION
**Pull request by natalieparry**
Restoring the .lib build for the win32 dll. This is different from the static version of the entire lib and serves as an import library for the dll that is important for c++ projects trying to link the lib. See issue #979 

Validation: Built the samplecpp project against the new import lib/dll. The project ran without reporting any linking failures.

### CODE REVIEW

File: Solutions/win32-dll/win32-dll.vcxproj [hunk 1]
```diff
@@ -91,7 +91,17 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">
   </ImportGroup>
-  <ImportGroup Label="Shared" />
+  <ImportGroup Label="Shared">
+	<Import Project="..\Clienttelemetry\Clienttelemetry.vcxitems" Label="Shared" />
```
Comment (Diego-Perez-Botero) [inline Solutions/win32-dll/win32-dll.vcxproj:95]: >	 [](http://example.com/codeflow?start=0&length=1)

Nit: please use whitespace instead of TABs

**Comment by Diego-Perez-Botero** [comment]
Please include validation steps in the PR description. Ideally, confirm that the following sample works for the DLL case: [https://github.com/microsoft/cpp_client_telemetry/tree/main/examples/cpp/SampleCpp](https://github.com/microsoft/cpp_client_telemetry/tree/main/examples/cpp/SampleCpp)

**Review by Diego-Perez-Botero — CHANGES_REQUESTED** [review]
:clock1:

**Review by Diego-Perez-Botero — APPROVED** [review]
:shipit:

File: examples/cpp/SampleCpp/SampleCpp.vcxproj [hunk 2]
```diff
@@ -87,39 +87,46 @@
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
     <CharacterSet>Unicode</CharacterSet>
+    <PlatformToolset>v142</PlatformToolset>
```
Comment (lalitb) [inline examples/cpp/SampleCpp/SampleCpp.vcxproj:90]: why explicit retarget to v142 is required?

**Comment by lalitb** [comment]
tagging @mkoscumb for review as this reverts #852

File: examples/cpp/SampleCpp/SampleCpp.vcxproj (hunk 2 above)
Comment (natalieparry) [reply examples/cpp/SampleCpp/SampleCpp.vcxproj:90]: Whoops good catch! This is left over from VS automatically retargeting the project. I'll revert this.

**Review by lalitb — APPROVED** [review]
LGTM. Do we still need a reference to the win32-lib project here, or can be removed (as this was added through #852)

https://github.com/microsoft/cpp_client_telemetry/blob/7db8680e9a0ccd7335f98fc2e1bf92b70073981a/Solutions/win32-dll/win32-dll.vcxproj#L321

**Pull request merged**